net/tcp/tcp_seqno.c

```diff
@@ -142,6 +144,34 @@ uint32_t tcp_addsequence(FAR uint8_t *seqno, uint16_t len)
 
 void tcp_initsequence(FAR uint8_t *seqno)
 {
+  int ret;
+
+  /* Get a random TCP sequence number */
+
+  ret = getrandom(&g_tcpsequence, sizeof(uint32_t), 0);
```

Comment:
let's check g_tcpsequence isn't zero before getrandom

Reply:
@xiaoxiang781216 do you propose not to call `getrandom` if `g_tcpsequence` is non-zero? If yes, then all the next values will be
```
  g_tcpsequence = g_tcpsequence % 2000000000;
...
  if (g_tcpsequence < 1000000000)
    {
      g_tcpsequence += 1000000000;
    }
```

Reply:
@xiaoxiang781216 initially my idea was to do exactly that because g_tcpsequence is a global variable and I thought it could be changed and mess with some ongoing transmission. After analyzing I noticed it is not used in any other place. So, it even doesn't need to be a global variable. What you think? Should I move it to inside the initsequence function?